pkg/containerwatcher/v1/container_watcher.go

```diff
@@ -189,8 +201,15 @@ func CreateIGContainerWatcher(cfg config.Config, applicationProfileManager appli
 		metrics.ReportEvent(utils.ExecveEventType)
 		applicationProfileManager.ReportFileExec(k8sContainerID, path, event.Args)
 		relevancyManager.ReportFileExec(event.Runtime.ContainerID, k8sContainerID, path)
-		ruleManager.ReportFileExec(event)
-		malwareManager.ReportFileExec(k8sContainerID, event)
+		ruleManager.ReportEvent(utils.ExecveEventType, &event)
+		malwareManager.ReportEvent(utils.ExecveEventType, &event)
+
+		// Report exec events to event receivers
+		if thirdPartyEventReceivers != nil && thirdPartyEventReceivers.Has(utils.ExecveEventType) {
```

Comment:
can you put this logic into a method like thirdPartyEventReceivers.ReportEvent() ?

Reply:
done